packages/react/src/components/Reaction/Reaction.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+import {
+  UIComponent,
+  childrenExist,
+  UIComponentProps,
+  ChildrenComponentProps,
+  commonPropTypes,
+  rtlTextContainer,
+  customPropTypes,
+  createShorthandFactory,
+} from '../../lib'
+import { Accessibility } from '../../lib/accessibility/types'
+import { defaultBehavior } from '../../lib/accessibility'
+import { ReactProps, ShorthandValue } from '../../types'
+import Icon from '../Icon/Icon'
+import Text from '../Text/Text'
+
+export interface ReactionSlotClassNames {
+  icon: string
+  count: string
+}
+
+export interface ReactionProps extends UIComponentProps<ReactionProps>, ChildrenComponentProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   * @default defaultBehavior
+   */
+  accessibility?: Accessibility
+
+  /** A reaction can have icon for the indicator of the reaction. */
+  icon?: ShorthandValue
+
+  /** A reaction can have count for the number of reaction. */
+  count?: number | string
+}
+
+/**
+ * A segment is used to create a grouping of related content.
+ */
+class Reaction extends UIComponent<ReactProps<ReactionProps>, any> {
+  static create: Function
+
+  static className = 'ui-reaction'
+
+  static slotClassNames: ReactionSlotClassNames
+
+  static displayName = 'Reaction'
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      content: false,
+    }),
+    icon: customPropTypes.itemShorthand,
+    count: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
```

Comment:
Should be `count` just a regular `content`? I don't a big difference here.

Reply:
I thought it would be awkward having number as a type for the content.. In the context of Reaction, count made more sense in my opinion. If you have strong opinion on this, I may change it..

Reply:
I'm not sure that we should be very strict there, another case is the localisation:

```jsx
<Reaction icon="thumbs down" count="1.2K" /> // fine
<Reaction icon="thumbs down" count={<Trans count='1.2' />} />// bomb
<Reaction icon="thumbs down" content={<Trans count='1.2' />} />// works and still consistent
```

Reply:
Agreed, let's keep consistency here, it can be shorthand as every other slot..